src/geometry.cpp

```diff
@@ -381,8 +388,13 @@ BoundaryInfo distance_to_boundary(Particle* p)
           lattice_distance = lat.distance(r, u, i_xyz);
           break;
         case LatticeType::hex:
-          Position r_hex {p->coord_[i-1].r.x, p->coord_[i-1].r.y,
-                          p->coord_[i].r.z};
+          auto& cell_above {model::cells[p->coord_[i-1].cell]};
+          Position r_hex {p->coord_[i-1].r};
+          r_hex -= cell_above->translation_;
+          if (coord.rotated) {
+            r_hex = r_hex.rotate(cell_above->rotation_);
```

Comment:
I'd rather not add more information to the `LocalCoord` struct. The rotation vector is 12 elements long so it would beef up that struct substantially. I'm hoping that we can refactor `HexLattice::distance` to just use the normal `r` from the same `coord` (as is done with rectangular lattices) which would avoid all this extra logic and extra translation/rotation.

Reply:
That sounds like a better option. Sounds good!